src/backend/CommandBufferStateTracker.cpp

```diff
@@ -542,10 +549,20 @@ namespace backend {
         return true;
     }
 
+    void CommandBufferStateTracker::ResetDependentPipelineApsects() {
+        constexpr ValidationAspects pipelineDependentAspects =
+            1 << VALIDATION_ASPECT_RENDER_PIPELINE |
+            1 << VALIDATION_ASPECT_COMPUTE_PIPELINE |
+            1 << VALIDATION_ASPECT_BIND_GROUPS |
+            1 << VALIDATION_ASPECT_VERTEX_BUFFERS |
+            1 << VALIDATION_ASPECT_INDEX_BUFFER;
+        aspects &= ~pipelineDependentAspects;
+    }
+
     bool CommandBufferStateTracker::SetPipelineCommon(PipelineBase* pipeline) {
         PipelineLayoutBase* layout = pipeline->GetLayout();
 
-        aspects.reset(VALIDATION_ASPECT_BIND_GROUPS);
+        ResetDependentPipelineApsects();
```

Comment:
Yeah VERTEX_BUFFERS would be in addition to the BIND_GROUPS currently present in the code.